rust/src/transformations/make_stable_expr/expr_sum/mod.rs

```diff
@@ -0,0 +1,308 @@
+use crate::core::{ExprFunction, Function, MetricSpace, StabilityMap, Transformation};
+use crate::domains::{ExprDomain, Margin, MarginPub, NumericDataType, SeriesDomain};
+use crate::error::*;
+use crate::metrics::{IntDistance, LpDistance, PartitionDistance};
+use crate::traits::{
+    AlertingAbs, CheckAtom, ExactIntCast, InfAdd, InfCast, InfMul, InfSqrt, InfSub, Number,
+    ProductOrd,
+};
+use crate::transformations::traits::UnboundedMetric;
+use crate::transformations::{CanFloatSumOverflow, CanIntSumOverflow, Sequential, SumRelaxation};
+use num::Zero;
+use polars::prelude::*;
+use std::collections::HashMap;
+
+use super::StableExpr;
+
+/// Polars operator to sum the active column in a LazyFrame
+///
+/// | input_metric                              |
+/// | ----------------------------------------- |
+/// | `PartitionDistance<SymmetricDistance>`    |
+/// | `PartitionDistance<InsertDeleteDistance>` |
+///
+/// # Arguments
+/// * `input_domain` - ExprDomain
+/// * `input_metric` - valid selections shown in table above
+/// * `expr` - an expression ending with sum
+pub fn make_expr_sum<MI, const P: usize>(
+    input_domain: ExprDomain,
+    input_metric: PartitionDistance<MI>,
+    expr: Expr,
+) -> Fallible<Transformation<ExprDomain, ExprDomain, PartitionDistance<MI>, LpDistance<P, f64>>>
+where
+    MI: 'static + UnboundedMetric,
+    (ExprDomain, PartitionDistance<MI>): MetricSpace,
+    Expr: StableExpr<PartitionDistance<MI>, PartitionDistance<MI>>,
+{
+    let Expr::Agg(AggExpr::Sum(prior_expr)) = expr else {
+        return fallible!(MakeTransformation, "expected function expression");
```

Comment:
Is the message accurate? It looks like you're matching for a `Sum` more particularly.